### PR TITLE
Add objects with meta data date column support

### PR DESCRIPTION
**Pull request by mbolka**
## Add date type info objects metadata and multihref with metadata.
This pull requests adds new type into supported types of meta columns.
image
image
image

### CODE REVIEW

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/multihrefMetadata.js [hunk 1]
```diff
@@ -424,6 +424,7 @@ pimcore.object.classes.data.multihrefMetadata = Class.create(pimcore.object.clas
             var types = {
                 number: t("objectsMetadata_type_number"),
                 text: t("objectsMetadata_type_text"),
+                date: t("objectsMetadata_type_date"),
```
Comment (brusch) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/multihrefMetadata.js]: The translation key `objectsMetadata_type_date` doesn't exist. 
Please use the `date` translation key instead, which does exist already. 
The other translations keys which are prefixed with `objectsMetadata_type_` are kind of legacy and are just aliased here: https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/bundles/CoreBundle/Resources/translations/aliases.json#L25 so there's nothing wrong by just usinf `date` directly´.

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/objectsMetadata.js [hunk 2]
```diff
@@ -248,6 +248,7 @@ pimcore.object.classes.data.objectsMetadata = Class.create(pimcore.object.classe
             var types = {
                 number: t("objectsMetadata_type_number"),
                 text: t("objectsMetadata_type_text"),
+                date: t("objectsMetadata_type_date"),
```
Comment (brusch) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/objectsMetadata.js]: Here the same please ;)

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/multihrefMetadata.js [hunk 3]
```diff
@@ -114,6 +114,10 @@ pimcore.object.tags.multihrefMetadata = Class.create(pimcore.object.tags.abstrac
                 cellEditor = function() {
                     return new Ext.form.TextField({});
                 };
+            } else if (this.fieldConfig.columns[i].type == "date" && !readOnly) {
+                cellEditor = function() {
+                    return new Ext.form.DateField({});
```
Comment (brusch) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/tags/multihrefMetadata.js]: Imho we should configure a `submitFormat` here, to ensure the format is always the same.

**Comment by mbolka** [comment]
@brusch All fixed. I have some doubts about it - values that are read from DB are displayed properly but are not assigned into Ext.form.data (date picked on first calendar open is a current date).
I have no idea how to fix it.
image
image

**Comment by brusch** [comment]
@mbolka there's still something wrong, when I add new relation items, it looks like that: 

image

**Comment by brusch** [comment]
@mbolka you should also try to solve the preselected date issue from above, otherwise I can't merge it. 
Thanks!

**Comment by brusch** [comment]
Closed due to inactivity. 
We can reopen this PR any time if needed and when the conflicts and missing issues are resolved.

**Pull request closed without merging**